Lib/test/test_typing.py

```diff
@@ -2456,6 +2456,12 @@ def foo(a: tuple[ForwardRef('T')]):
         self.assertEqual(get_type_hints(foo, globals(), locals()),
                          {'a': tuple[T]})
 
+    def test_double_forward(self):
+        def foo(a: 'List[\'int\']'):
```

Comment:
I was about to approve this PR when I realized that we should make this work with `list` -- I tried and it doesn't currently.

Reply:
You're right concerning the fact that it doesn't work with `list`, but I've found that it's not related to this PR; actually, I think there is a significant defect in current PEP 585 implementation. In fact:
``` python
from dataclasses import dataclass, field
from typing import get_type_hints, List, ForwardRef

@dataclass
class Node:
    children: list["Node"] = field(default_factory=list)
    children2: List["Node"] = field(default_factory=list)

assert get_type_hints(Node) == {"children": list["Node"], "children2": List[Node]}
assert List["Node"].__args__ == (ForwardRef("Node"),)
assert list["Node"].__args__ == ("Node",)  # No ForwardRef here, so no evaluation by get_type_hints
```
I will open an issue on *bugs.python.org* to discuss about this, because I think this PR is not the best place to do it.

Reply:
Heh, you're right, thanks for the analysis. We probably will not ever support this: importing `ForwardRef` from the built-in generic alias code would be problematic, and once `from __future__ import annotations` is always on there's no need to quote the argument anyway. Sorry for the hiccup. I will now approve your PR.